components/support/crypto/nss-sys/src/backends/dylib_nss.rs

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::types::nspr::*;
+use crate::types::nss3::*;
+use std::os::raw::{c_char, c_uchar};
+
+#[link(name = "nss3")]
+#[link(name = "nspr4")]
+extern "C" {
+    pub fn PK11_HashBuf(
+        hashAlg: SECOidTag,
+        out: *mut c_uchar,
+        r#in: *const c_uchar,
+        len: PRInt32,
+    ) -> SECStatus;
+    pub fn NSS_NoDB_Init(configdir: *const c_char) -> SECStatus;
+    pub fn NSS_InitContext(
+        configdir: *const c_char,
+        certPrefix: *const c_char,
+        keyPrefix: *const c_char,
+        secmodName: *const c_char,
+        initParams: *mut NSSInitParameters,
+        flags: PRUint32,
+    ) -> *mut NSSInitContext;
+    pub fn NSS_IsInitialized() -> PRBool;
+    pub fn NSS_GetVersion() -> *const c_char;
+    pub fn PR_GetError() -> PRErrorCode;
+    pub fn PR_GetErrorTextLength() -> PRInt32;
+    pub fn PR_GetErrorText(out: *mut c_uchar) -> PRInt32;
```

Comment:
Is there any way we can safe the type signatures of these functions between `dylib_nss.rs` and `geckoview_nss.rs`, so that we only have to audit and maintain a single copy?

Reply:
build.rs shenanigans. A macro with different definitions on the different platforms.

No clean way, though. Although the macro way could be good so long as it's done without obscuring what's happening too much.

Reply:
I've implemented the `nss_exports` macro!